vm.c

```diff
@@ -1415,7 +1415,7 @@ rb_proc_ractor_make_shareable(VALUE self)
         proc->is_isolated = TRUE;
     }
 
-    FL_SET_RAW(self, RUBY_FL_SHAREABLE);
+    rb_obj_freeze(self);
```

Comment:
does it marks shareable flag on a Proc?

Reply:
It does after traversing through the objects as part of `rb_ractor_make_shareable`. Setting it here before was allowing unfrozen objects to leak through.

Reply:
Ah, ok, `mark_shareable` will set.
Strictly speaking the Proc is not needed to be frozen, but okay.
no problem to merge.

Reply:
I think we need to freeze to prevent instance variables from being set after it's made shareable.

Reply:
Yes it is safer. But for example Ractor objects are shareable but not frozen. The instance variables can be accessed only from main ractor.

Reply:
Ah! I see, thank you. I didn't see that behaviour in `generic_ivtbl`